### PR TITLE
Fixes for conmon support on FreeBSD

### PR DESCRIPTION
**Pull request by dfr**
This PR fixes two small problems in the conmon support code:

- Moving linux-specific capabilites code from oci_conmon_exec_common.go to the platform-specific file. This fixes a problem that stopped 'podman exec' from working on FreeBSD.
- Filtering out a confusing (but harmless) error which happened on FreeBSD when the container stdin closes.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by mheon** [comment]
LGTM, but you have lint issues

File: libpod/oci_conmon_exec_common.go [hunk 1]
```diff
@@ -745,34 +744,9 @@ func prepareProcessExec(c *Container, options *ExecOptions, env []string, sessio
...
 	}
 
-	ctrSpec, err := c.specFromState()
-	if err != nil {
-		return nil, err
-	}
-
-	allCaps, err := capabilities.BoundingSet()
-	if err != nil {
+	if err := setProcessCapabilitiesExec(c, options, user, execUser, pspec); err != nil {
```
Comment (rhatdan) [inline libpod/oci_conmon_exec_common.go:747]: I would prefer this to me.
c.setProcessCapabilitiesExec(options, user, execUser, pspec)

Comment (dfr) [reply libpod/oci_conmon_exec_common.go]: Done. I also changed the signature of prepareProcessExec for consistency.

**Comment by rhatdan** [comment]
LGTM
@vrothberg @giuseppe @flouthoc PTAL

**Review by vrothberg — COMMENTED** [review]
/lgtm
/hold

**Review by flouthoc — APPROVED** [review]
LGTM
/hold cancel

**Pull request merged**